app/models/miq_report/generator.rb

```diff
@@ -191,127 +191,141 @@ def generate_table(options = {})
-      targets = db_class
-      targets = db_class.find_entries(ext_options) if targets.respond_to?(:find_entries)
-      # TODO: add once only_cols is fixed
-      # targets = targets.select(only_cols)
-      where_clause = MiqExpression.merge_where_clauses(self.where_clause, options[:where_clause])
-      ## add in virtual attributes that can be calculated from sql
-      va_sql_cols = cols.select do |col|
-        db_class.virtual_attribute?(col) && db_class.attribute_supported_by_sql?(col)
-      end
-      rbac_opts = options.merge(
-        :targets          => targets,
-        :filter           => conditions,
-        :include_for_find => includes,
-        :where_clause     => where_clause,
-        :skip_counts      => true,
+      results, extras[:group_by_tag_cols], extras[:group_by_tags] = db_class.group_by_tags(
+        db_class.find_entries(ext_options).where(where_clause).where(options[:where_clause]),
+        :category  => performance[:group_by_category],
+        :cat_model => options[:cat_model],
+        :include   => get_include_for_find
       )
-
-      rbac_opts[:extra_cols] = va_sql_cols unless va_sql_cols.nil? || va_sql_cols.empty?
-
-      results, attrs = Rbac.search(rbac_opts)
-      results = Metric::Helper.remove_duplicate_timestamps(results)
-      results = BottleneckEvent.remove_duplicate_find_results(results) if db == "BottleneckEvent"
-      @user_categories = attrs[:user_filters]["managed"]
+      build_correlate_tag_cols
     end
+    results
+  end
 
-    if db_options && db_options[:long_term_averages] && results.first.kind_of?(MetricRollup)
-      # Calculate long_term_averages and save in extras
-      self.extras[:long_term_averages] = Metric::LongTermAverages.get_averages_over_time_period(results.first.resource, db_options[:long_term_averages].merge(:ext_options => ext_options))
-    end
+  # Ad-hoc daily performance reports
+  #   Daily for: Performance - Clusters...
+  def generate_daily_metric_rollup_results(options = {})
+    unless conditions.nil?
+      conditions.preprocess_options = {:vim_performance_daily_adhoc => (time_profile && time_profile.rollup_daily_metrics)}
+      exp_sql, exp_includes = conditions.to_sql
+      # only_cols += conditions.columns_for_sql # Add cols references in expression to ensure they are present for evaluation
+    end
+
+    time_range = Metric::Helper.time_range_from_offset(interval, db_options[:start_offset], db_options[:end_offset], tz)
+    # TODO: add .select(only_cols)
+    db_includes = get_include_for_find
+    results = Metric::Helper.find_for_interval_name('daily', time_profile || tz, db_klass)
+                            .where(where_clause).where(exp_sql)
+                            .where(options[:where_clause])
+                            .where(:timestamp => time_range)
+                            .includes(db_includes)
+                            .references(db_includes)
```

Comment:
seems like we would be concerned about this `references` call

Reply:
Yeah, though it was pulled from this line:

https://github.com/ManageIQ/manageiq/pull/17189/files#diff-86181c45401db85fe3c8ebdffba65b28L253

So I was simply moving this around.